x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/planner/PlannerUtils.java

```diff
@@ -72,6 +81,35 @@ public static Tuple<PhysicalPlan, PhysicalPlan> breakPlanBetweenCoordinatorAndDa
         return new Tuple<>(coordinatorPlan, dataNodePlan.get());
     }
 
+    public static PhysicalPlan dataNodeReductionPlan(LogicalPlan plan, PhysicalPlan unused) {
```

Comment:
It would be worth a comment about "unused" being eventually replaced (as "unused" is used for not "touched anymore" params, usually)

Reply:
This `unused` is to satisfy the TopNExec node that gets created since it must have a child node. This one will get replaced afterwards on each data node with an ExchangeSourceExec (the node level reduction layer).

Reply:
I meant to add the above as a code comment. Since "unused" is actually used (even if as a place holder). Anyways, nit, can ofc. remain as is or touched on any related change later.